aidial_adapter_openai/images.py

```diff
@@ -0,0 +1,134 @@
+from typing import Any, AsyncGenerator, Optional
+
+import aiohttp
+from fastapi.responses import JSONResponse, Response, StreamingResponse
+
+from aidial_adapter_openai.utils.exceptions import HTTPException
+from aidial_adapter_openai.utils.storage import FileStorage, upload_base64_file
+from aidial_adapter_openai.utils.streaming import (
+    END_CHUNK,
+    build_chunk,
+    chunk_format,
+    generate_id,
+)
+
+
+async def generate_image(api_url: str, api_key: str, user_prompt: str) -> Any:
+    async with aiohttp.ClientSession() as session:
+        async with session.post(
+            api_url,
+            json={"prompt": user_prompt, "response_format": "b64_json"},
+            headers={"api-key": api_key},
+        ) as response:
+            status_code = response.status
+
+            data = await response.json()
+
+            if status_code == 200:
+                return data
+            else:
+                return JSONResponse(content=data, status_code=status_code)
+
+
+def build_custom_content(base64_image: str, revised_prompt: str) -> Any:
+    return {
+        "custom_content": {
+            "attachments": [
+                {"title": "Image", "type": "image/png", "data": base64_image},
+                {"title": "Revised prompt", "data": revised_prompt},
+            ]
+        }
+    }
+
+
+async def generate_stream(
+    id: str, created: str, custom_content: Any
+) -> AsyncGenerator[Any, Any]:
+    yield chunk_format(
+        build_chunk(id, None, {"role": "assistant"}, created, True)
+    )
+
+    yield chunk_format(build_chunk(id, None, custom_content, created, True))
+
+    yield chunk_format(
```

Comment:
You could save a few lines of code if you pull `usage` into to a dedicated variable and then use it in `build_chunk`, so that it will be formatted as a single line.

Reply:
agree, created `IMG_USAGE` constant